app/import_validators.go

```diff
@@ -560,24 +560,30 @@ func validateDirectPostImportData(data *DirectPostImportData, maxPostSize int) *
 	return nil
 }
 
-func validateEmojiImportData(data *EmojiImportData) *model.AppError {
+// validateEmojiImportData validates emoji data and returns if the import name
+// conflicts with a system emoji.
+func validateEmojiImportData(data *EmojiImportData) (bool, *model.AppError) {
```

Comment:
This seems to be a slightly cumbersome way to achieve it. IMO, a more idiomatic way would be to use the error value to detect a system emoji case and appropriately log it.

So essentially, instead of clubbing the system emoji check with the other conditions (` || inSystemEmoji(name)`), what we could do is split that out to a separate check, with a separate AppError ID.

```go
if inSystemEmoji(name) {
	return NewAppError("Emoji.IsValid", "some_other_name", nil, "", http.StatusBadRequest)	
}
```

And then in `importEmoji`, we could check if `err.Id == "some_other_name"` and then just skip with a warning. This way, we don't need to add an additional bool variable to differentiate between the case, and `InSystemEmoji` can remain unexported. And this also gives us the flexibility to keep adding more edge-cases by using errors, without returning more values.